monitor/extensions/csv-datastore-wizard/src/main/java/org/datacleaner/monitor/server/wizard/CsvDatastoreWizardSession.java

```diff
@@ -86,26 +88,31 @@ protected WizardPageController nextPageControllerExisting() {
                 return new CsvDatastoreLocationWizardPage(getWizardContext(), "my_file.csv", false) {
 
                     @Override
-                    protected WizardPageController nextPageController(String filepath, File file) {
-                        if (!filepath.toLowerCase().endsWith(".csv")) {
-                            if (!filepath.toLowerCase().endsWith(".tsv")) {
-                                if (!filepath.toLowerCase().endsWith(".txt")) {
-                                    // only .csv and .tsv files are allowed to
-                                    // be referenced on the server, for security
-                                    // reasons.
-                                    throw new DCUserInputException(
-                                            "For security reasons, only existing .csv, .tsv or .txt files can be referenced on the server");
+                    protected WizardPageController nextPageController(Resource resource) {
+                        if (resource instanceof FileResource) {
+                            final File file = ((FileResource) resource).getFile();
+                            final String filepath = file.getAbsolutePath();
+                            if (!filepath.toLowerCase().endsWith(".csv")) {
```

Comment:
I will improve the with the "OR" and check about the extensions. Why the special cases ? I have no idea.